dlp/api/Snippets/ReidentifyFreeTextFpe.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// [START dlp_reidentify_text_fpe]
+
+using System;
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.Dlp.V2;
+using Google.Protobuf;
+using static Google.Cloud.Dlp.V2.CryptoReplaceFfxFpeConfig.Types;
+
+public class ReidentifyFreeTextFpe
+{
+    public static ReidentifyContentResponse ReidentifyContent(
+        string projectId,
+        string text,
+        string keyName,
+        string wrappedKey,
+        FfxCommonNativeAlphabet alphabet = FfxCommonNativeAlphabet.Numeric,
+        InfoType surrogateType = null)
+    {
+        // Instantiate the client.
+        var dlp = DlpServiceClient.Create();
+
+        // Construct the infoType which will be used as both surrogate type and infoType.
+        var infotype = surrogateType ?? new InfoType { Name = "PHONE_TOKEN" };
+
+        // Specify how to un-encrypt the previously de-identified information.
+        var cryptoReplaceFfxFpeConfig = new CryptoReplaceFfxFpeConfig
+        {
+            CryptoKey = new CryptoKey
+            {
+                KmsWrapped = new KmsWrappedCryptoKey
+                {
+                    CryptoKeyName = keyName,
+                    WrappedKey = ByteString.FromBase64(wrappedKey)
+                }
+            },
+            CommonAlphabet = alphabet,
+            SurrogateInfoType = infotype
+        };
+
+        // Construct re-identify config using CryptoReplaceFfxFpeConfig transformation method.
```

Comment:
Sorry, a few more edits here:

Construct the re-identify config using the CryptoReplaceFfxFpeConfig transformation method.

Reply:
Done thanks.